### PR TITLE
Classification and regression evaluation notebooks

### PR DESCRIPTION
**Pull request by hhpang**
Classification notebook with example provided, and regression notebook with example provided. Training time: <15-20min each

### CODE REVIEW

**Review by yamini — APPROVED** [review]
🚢

File: docs/notebooks/downstream_machine_learning_classification_evaluation.ipynb [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Evaluate synthetic data on downstream classification models\n",
+    "\n",
+    "### How to use this notebook\n",
+    "Use this notebook to understand the quality and performance of your synthetic or augmented data on downstream machine learning classification tasks. \n",
```
Comment (kboyd) [inline docs/notebooks/downstream_machine_learning_classification_evaluation.ipynb:11]: suggesiton: Can we be more explicit that we're comparing synthetic to real data performance here? Real data isn't mentioned at all right now.

And did we decide on terminology for "synthetic" or "synthetic or augmented", I think I saw a discussion but didn't catch the resolution.

File: docs/notebooks/downstream_machine_learning_classification_evaluation.ipynb [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+    "\n",
+    "### Installation\n",
+    "Install Gretel Client to use both Gretel's synthetic models as well as the Gretel Evaluate Classification model. You'll have to get your API key from the [Gretel console](https://www.console.gretel.ai) to configure your session. "
```
Comment (kboyd) [inline docs/notebooks/downstream_machine_learning_classification_evaluation.ipynb:14]:
```suggestion
    "Install the Gretel Client to use both Gretel's synthetic models as well as the Gretel Evaluate Classification model. You'll have to get your API key from the [Gretel console](https://www.console.gretel.ai) to configure your session. "
```

File: docs/notebooks/downstream_machine_learning_classification_evaluation.ipynb [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    " ## Try: Generate synthetic data, then evaluate the synthetic data on classifiers against real-world data\n",
+    " First, we'll generate synthetic data using a publicly available bank marketing dataset, which predicts whether a client will subscribe a term deposite (prediction: yes/no in column \"y\"). We'll use Gretel's LSTM model to train on the real-world and generate the synthetic data.\n",
```
Comment (kboyd) [inline docs/notebooks/downstream_machine_learning_classification_evaluation.ipynb:44]:
```suggestion
    " First, we'll generate synthetic data using a publicly available bank marketing dataset. The target variable is if a client will subscribe to a term deposit (prediction: yes/no in column \"y\"). We'll use Gretel's LSTM model to train on the real-world data and then generate synthetic data from the model.\n",
```

Comment (kboyd) [inline docs/notebooks/downstream_machine_learning_classification_evaluation.ipynb:44]: question: Do we want to give credit to the source of the data set (I'm guessing this is the bank datsaset from UCI ML repository).

File: docs/notebooks/downstream_machine_learning_classification_evaluation.ipynb [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+    " \n",
+    " To use the Gretel Evaluate Classification model, you must indicate the target column. Optionally, you can change the test-holdout amount, which is a float indicating the amount of real-world data you want to use as a holdout for testing the downstream classifiers. Youc an also optionally select which classifiers to use and which metric to optimize for. "
```
Comment (kboyd) [inline docs/notebooks/downstream_machine_learning_classification_evaluation.ipynb:46]:
```suggestion
    " To use the Gretel Evaluate Classification model, you must indicate the target column. Optionally, you can change the test-holdout amount, which is a float indicating the amount of real-world data you want to use as a holdout for testing the downstream classifiers. You can also optionally select which classifiers to use and which metric to optimize for. "
```

File: docs/notebooks/downstream_machine_learning_classification_evaluation.ipynb [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+    "evaluate.peek()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Results\n",
+    "To see the results of your evaluation in the Gretel Console, go to your [Projects list](https://console-staging.gretel.ai/projects) and look for the projects titled \"bank-marketing-classification-notebook\" or \"evaluate-bank-classification-notebook-2\". You can download the data quality report, regression report, and check out details of the model configuration from there.  \n",
```
Comment (kboyd) [inline docs/notebooks/downstream_machine_learning_classification_evaluation.ipynb:289]:
```suggestion
    "To see the results of your evaluation in the Gretel Console, go to your [Projects list](https://console-staging.gretel.ai/projects) and look for the projects titled \"bank-marketing-classification-notebook\" or \"evaluate-bank-classification-notebook-2\". You can download the data quality report, classification report, and check out details of the model configuration from there.  \n",
```

File: docs/notebooks/downstream_machine_learning_regression_evaluation.ipynb [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Evaluate synthetic data on downstream regression models\n",
+    "\n",
+    "### How to use this notebook\n",
+    "Use this notebook to understand the quality and performance of your synthetic or augmented data on downstream machine learning regression tasks. \n",
+    "\n",
```
Comment (kboyd) [inline docs/notebooks/downstream_machine_learning_regression_evaluation.ipynb:12]: suggestion: similar to classification, mention comparison to real world data here.

File: docs/notebooks/downstream_machine_learning_regression_evaluation.ipynb [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+    "### Installation\n",
+    "Install Gretel Client to use Gretel's synthetic models as well as the Gretel Evaluate Regression model. You'll have to get your API key from the [Gretel console](https://www.console.gretel.ai) to configure your session. "
```
Comment (kboyd) [inline docs/notebooks/downstream_machine_learning_regression_evaluation.ipynb:14]:
```suggestion
    "Install the Gretel Client to use Gretel's synthetic models as well as the Gretel Evaluate Regression model. You'll have to get your API key from the [Gretel console](https://www.console.gretel.ai) to configure your session. "
```

File: docs/notebooks/downstream_machine_learning_regression_evaluation.ipynb [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+    "configure_session(endpoint=\"https://api.gretel.cloud\", api_key=\"prompt\", cache=\"yes\")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    " ## Try: Generate synthetic data, then evaluate the synthetic data on regression models against real-world data\n",
+    " First, we'll generate synthetic data using a publicly available [processed Cleveland Clinic heart disease dataset](https://archive.ics.uci.edu/ml/datasets/heart+disease), which predicts the likelihood of heart disease present in a patient. We'll use Gretel's LSTM model to train on the real-world data and generate the synthetic data. For the regression evaluation, we'll predict the heart rate value in this target column: \"0\".\n",
```
Comment (kboyd) [inline docs/notebooks/downstream_machine_learning_regression_evaluation.ipynb:43]: suggestion: Would be nice to add column names to this dataset that we have in AWS. There are somewhat more interpretable column names at https://archive.ics.uci.edu/ml/datasets/heart+disease. Something more descriptive than "0" would look a bit more polished I think. Even "target" or something generic be an improvement in my opinion. And giving some sort of name to the columns also makes the SQS report that will be available for the synthetic data a lot more understandable.

File: docs/notebooks/downstream_machine_learning_regression_evaluation.ipynb [hunk 9]
```diff
@@ -0,0 +1,311 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from gretel_client.helpers import poll\n",
+    "from gretel_client.projects.models import read_model_config\n",
+    "\n",
+    "# We'll import the Dow Jones stock price dataset from Gretel's public S3 bucket\n",
```
Comment (kboyd) [inline docs/notebooks/downstream_machine_learning_regression_evaluation.ipynb:127]: issue: comment does not match the dataset we are loading.

**Review by kboyd — APPROVED** [review]
A few small recommended edits and such, but looks good overall.

File: docs/notebooks/downstream_machine_learning_classification_evaluation.ipynb (hunk 3 above)
Comment (hhpang) [reply docs/notebooks/downstream_machine_learning_classification_evaluation.ipynb]: yup thanks for catching!

File: docs/notebooks/downstream_machine_learning_regression_evaluation.ipynb (hunk 9 above)
Comment (hhpang) [reply docs/notebooks/downstream_machine_learning_regression_evaluation.ipynb:127]: ty for catching this

File: docs/notebooks/downstream_machine_learning_regression_evaluation.ipynb (hunk 6 above)
Comment (hhpang) [reply docs/notebooks/downstream_machine_learning_regression_evaluation.ipynb:17]: added

File: docs/notebooks/downstream_machine_learning_regression_evaluation.ipynb (hunk 8 above)
Comment (hhpang) [reply docs/notebooks/downstream_machine_learning_regression_evaluation.ipynb:48]: that's a great suggestion! i'd like to open a separate ticket for adding descriptive names for all benchmark/gretel datasets missing headers - will have to cut out some time for this, there are a few missing

Comment (hhpang) [reply docs/notebooks/downstream_machine_learning_regression_evaluation.ipynb]: we updated the datasets in benchmark, so this has been changed in notebook to be more descriptive

**Pull request merged**